plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/xpl/ImageName.java

```diff
@@ -0,0 +1,198 @@
+ *     <li>consol/tomcat-8.0</li>
+ *     <li>consol/tomcat-8.0:8.0.9</li>
+ *     <li>docker.consol.de:5000/tomcat-8.0</li>
+ *     <li>docker.consol.de:5000/jolokia/tomcat-8.0:8.0.9</li>
+ * </ul>
+ *
+ * @author roland
+ * @since 22.07.14
+ */
+public class ImageName {
+
+    // The repository part of the full image
+    private String repository;
+
+    // Registry
+    private String registry;
+
+    // Tag name
+    private String tag;
+
+    // User name
+    private String user;
+
+    /**
+     * Create an image name
+     *
+     * @param fullName The fullname of the image in Docker format.
+     */
+    public ImageName(String fullName) {
+        this(fullName,null);
+    }
+
+    /**
+     * Create an image name with a tag. If a tag is provided (i.e. is not null) then this tag is used.
+     * Otherwise the tag of the provided name is used (if any).
+     *
+     * @param fullName The fullname of the image in Docker format. I
+     * @param givenTag tag to use. Can be null in which case the tag specified in fullName is used.
+     */
+    public ImageName(String fullName,String givenTag) {
+        if (fullName == null) {
+            throw new NullPointerException("Image name must not be null");
+        }
+        Pattern tagPattern = Pattern.compile("^(.+?)(?::([^:/]+))?$");
+        Matcher matcher = tagPattern.matcher(fullName);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException(fullName + " is not a proper image name ([registry/][repo][:port]");
+        }
+        tag = givenTag != null ?
+                givenTag :
+                matcher.groupCount() > 1 ? matcher.group(2) : null;
+        String rest = matcher.group(1);
+
+        String[] parts = rest.split("\\s*/\\s*");
+        if (parts.length == 1) {
+            registry = null;
+            user = null;
+            repository = parts[0];
+        } else if (parts.length >= 2) {
+            if (isRegistry(parts[0])) {
+                registry = parts[0];
+                user = parts[1];
+                repository = joinTail(parts);
+            } else {
+                registry = null;
+                user = parts[0];
+                repository = rest;
+            }
+        }
+    }
+
+    public String getRepository() {
+        return repository;
+    }
+
+    public String getRegistry() {
+        return registry;
+    }
+
+    public String getTag() {
+        return tag;
+    }
+
+    public boolean hasRegistry() {
+        return registry != null && registry.length() > 0;
+    }
+
+    private String joinTail(String[] parts) {
+        StringBuilder builder = new StringBuilder();
+        for (int i = 1;i < parts.length; i++) {
+            builder.append(parts[i]);
+            if (i < parts.length - 1) {
+                builder.append("/");
+            }
+        }
+        return builder.toString();
+    }
+
+    private boolean isRegistry(String part) {
+        return part.contains(".") || part.contains(":");
```

Comment:
According to org.eclipse.linuxtools.internal.docker.core.DockerImage, repository and name also can have '.' whereas registry if present must have ':'.